v3/mqtt.go

```diff
@@ -496,28 +549,48 @@ func (client *MqttClient) receivePacket(ctx context.Context,
 // Implementation of the delegate for mqttConn to handle publish from the
 // server on topics that we've subscribed to. We only unpack it and send it
 // to the caller
-func (client *MqttClient) handlePubReceive(p *packet.PublishPacket) {
+func (client *MqttClient) handlePubReceive(p *packet.PublishPacket,
+	receiveTime time.Time) {
 	logInfo("[MQTT] received message for topic %s", p.Message.Topic)
 
-	pubData := MqttSubData{p.Message.Topic, p.Message.Payload}
+	pubData := MqttSubData{p.Message.Topic, p.Message.Payload, receiveTime}
```

Comment:
Omitting attribute name is not allowed by `go vet`. And I think it is better to explicit writing attribute name.